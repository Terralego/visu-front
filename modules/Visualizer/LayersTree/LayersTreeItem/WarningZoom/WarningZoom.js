import React from 'react';

import {
  Intent,
} from '@blueprintjs/core';
import translateMock from '../../../../../utils/translate';
import Tooltip from '../../../../../components/Tooltip';


export const WarningZoom = props => {
  const {
    display,
    isActive,
    minZoomLayer,
    children,
    translate = translateMock({
      'terralego.visualizer.layerstree.warningzoom.message': 'Visible from zoom {{minzoom}}',
    }),
  } = props;

  if (!display || !isActive) return children;
  return (
    <Tooltip
      className="layerstree-node-content__item-tooltip-warning"
      content={(<span>{translate('terralego.visualizer.layerstree.warningzoom.message', { minzoom: minZoomLayer })}</span>)}
      intent={Intent.WARNING}
      usePortal={false}
    >
      {children}
    </Tooltip>
  );
};

export default WarningZoom;
