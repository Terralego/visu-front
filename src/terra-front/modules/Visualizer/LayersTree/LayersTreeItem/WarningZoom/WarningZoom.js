import React from 'react';

import {
  Intent,
} from '@blueprintjs/core';
import translateMock from '../../../../../utils/translate';
import Tooltip from '../../../../../components/Tooltip';


export const WarningZoom = props => {
  const {
    display,
    minZoomLayer,
    children,
    translate = translateMock({
      'terralego.visualizer.layerstree.warningzoom.message': 'Visible from zoom {{minzoom}}',
    }),
  } = props;

  if (!display) return children;
  return (
    <Tooltip
      className="layerstree-node-content__item-tooltip-warning"
      content={(<span>{translate('terralego.visualizer.layerstree.warningzoom.message', { minzoom: minZoomLayer })}</span>)}
      intent={Intent.WARNING}
      usePortal={false}
    >
      {children}
    </Tooltip>
  );
};

export default WarningZoom;
