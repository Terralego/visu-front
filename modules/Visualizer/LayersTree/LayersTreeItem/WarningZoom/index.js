import WarningZoom from './WarningZoom';

export default WarningZoom;
