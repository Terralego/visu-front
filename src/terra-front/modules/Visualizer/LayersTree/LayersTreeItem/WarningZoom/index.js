import React from 'react';
import compose from '../../../../../utils/compose';
import { processWarningAccordingToZoom } from '../../../services/warningZoom';
import WarningZoom from './WarningZoom';
import { connectLayersTree } from '../../LayersTreeProvider/context';

const withWarningAccordingToZoom = WrappedComponent => props => {
  const { map, layer, isActive, children } = props;
  if (!isActive) return children;
  const { showWarning, minZoomLayer } = processWarningAccordingToZoom(map, layer);
  return (
    <WrappedComponent display={showWarning} minZoomLayer={minZoomLayer} {...props}>
      {children}
    </WrappedComponent>
  );
};

export default compose(
  withWarningAccordingToZoom,
  connectLayersTree('translate'),
)(WarningZoom);
