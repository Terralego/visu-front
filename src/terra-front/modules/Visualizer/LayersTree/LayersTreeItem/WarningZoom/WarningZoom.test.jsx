import React from 'react';
import renderer from 'react-test-renderer';

import { WarningZoom } from './WarningZoom';

jest.mock('@blueprintjs/core', () => ({
  Intent: {
    PRIMARY: 'primary',
  },
  Tooltip ({ children }) { return children; },
}));

it('should render with warning', () => {
  const tree = renderer.create((
    <WarningZoom
      display
      isActive
      minZoomLayer={14}
    >
      <p>foo</p>
    </WarningZoom>
  ));
  expect(tree.toJSON()).toMatchSnapshot();
});

it('should render without warning', () => {
  const tree = renderer.create((
    <WarningZoom
      display={false}
      isActive={false}
      minZoomLayer={14}
    >
      <p>foo</p>
    </WarningZoom>
  ));
  expect(tree.toJSON()).toMatchSnapshot();
});
