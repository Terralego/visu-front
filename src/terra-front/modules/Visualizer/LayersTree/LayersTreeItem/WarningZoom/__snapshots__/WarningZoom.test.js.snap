// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render with warning 1`] = `
<p>
  foo
</p>
`;

exports[`should render without warning 1`] = `
<p>
  foo
</p>
`;
